### PR TITLE
Added explicit idempotency detector

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Motivation and Context

Support for attribute `IdempotencyPredicate.IDEMPOTENT` was added in a special and now it's a normal idempotency detector, like any other.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: riptide-idempotency/src/main/java/org/zalando/riptide/idempotency/IdempotencyPredicate.java [hunk 1]
```diff
@@ -27,20 +30,27 @@
...
+        return decide(arguments) == Decision.ACCEPT;
+    }
+
+    private Decision decide(final RequestArguments arguments) {
+        final Test test = modified ->
+                arguments.equals(modified) ? NEUTRAL : decide(modified);
+
+        return detectors.stream()
+                .map(detector -> detector.test(arguments, test))
+                .max(naturalOrder())
```
Comment (lukasniemeier-zalando) [inline riptide-idempotency/src/main/java/org/zalando/riptide/idempotency/IdempotencyPredicate.java]: Ha trick 17. And now try to also do early exit on `DENY` :wink:

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: riptide-idempotency/src/main/java/org/zalando/riptide/idempotency/IdempotencyPredicate.java (hunk 1 above)
Comment (whiskeysierra) [reply riptide-idempotency/src/main/java/org/zalando/riptide/idempotency/IdempotencyPredicate.java]: Yeah, going to change that to a for-loop with an early break.

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**